pkg/apis/core/validation/validation.go

```diff
@@ -4688,6 +4688,15 @@ func ValidatePodEphemeralContainersUpdate(newPod, oldPod *core.Pod, opts PodVali
 	allErrs = append(allErrs, validatePodMetadataAndSpec(newPod, opts)...)
 	allErrs = append(allErrs, ValidatePodSpecificAnnotationUpdates(newPod, oldPod, fldPath.Child("annotations"), opts)...)
 
+	// static pods don't support ephemeral containers #113935
+	oldAnnotations := oldPod.Annotations
+	for k := range oldAnnotations {
```

Comment:
this is a map[string]string you don't have to iterate
```
	if _, isMirror := annotations[core.MirrorPodAnnotationKey]; isMirror {
allErrs = append(allErrs, field.Forbidden(fldPath.Child("annotations"), fmt.Sprintf("static pods (%q) do not support ephemeral containers", oldPod.ObjectMeta.GetName())))
	}
```

Reply:
dang it. obvious